bika/lims/browser/analyses/qc.py

```diff
@@ -36,81 +40,84 @@ class QCAnalysesView(AnalysesView):
-            "/++resource++bika.lims.images/referencesample.png"
-
-    # TODO-performance: Do not use object. Using brain, use meta_type in
-    # order to get the object's type
-    def folderitem(self, obj, item, index):
-        """Prepare a data item for the listing.
+        super(QCAnalysesView, self).__init__(context, request, **kwargs)
+
+        icon_path = "/++resource++bika.lims.images/referencesample.png"
+        self.icon = "{}{}".format(self.portal_url, icon_path)
 
-        :param obj: The catalog brain or content object
-        :param item: Listing item (dictionary)
-        :param index: Index of the listing item
-        :returns: Augmented listing data item
+    def update(self):
+        """Update hook
         """
+        super(AnalysesView, self).update()
 
-        obj = obj.getObject()
-        # Group items by RefSample - Worksheet - Position
-        ws = obj.getWorksheet()
-        wsid = ws and ws.id or ''
-        wshref = ws.absolute_url() or None
-        if wshref:
-            item['replace']['Worksheet'] = "<a href='%s'>%s</a>" % (
-                wshref, wsid)
-
-        imgtype = ""
-        if obj.portal_type == 'ReferenceAnalysis':
-            antype = QCANALYSIS_TYPES.getValue(obj.getReferenceType())
-            if obj.getReferenceType() == 'c':
-                imgtype = "<img title='%s' " \
-                          "src='%s/++resource++bika.lims.images/control.png" \
-                          "'/>&nbsp;" % (
-                              antype, self.context.absolute_url())
-            if obj.getReferenceType() == 'b':
-                imgtype = "<img title='%s' " \
-                          "src='%s/++resource++bika.lims.images/blank.png" \
-                          "'/>&nbsp;" % (
-                              antype, self.context.absolute_url())
-            item['replace']['Partition'] = "<a href='%s'>%s</a>" % (
-                obj.aq_parent.absolute_url(), obj.aq_parent.id)
-        elif obj.portal_type == 'DuplicateAnalysis':
-            antype = QCANALYSIS_TYPES.getValue('d')
-            imgtype = "<img title='%s' " \
-                      "src='%s/++resource++bika.lims.images/duplicate.png" \
-                      "'/>&nbsp;" % (
-                          antype, self.context.absolute_url())
-            item['sortcode'] = '%s_%s' % (obj.getRequestID(), obj.getKeyword())
-
-        item['before']['Service'] = imgtype
-        item['sortcode'] = '%s_%s' % (obj.getReferenceAnalysesGroupID(),
-                                      obj.getKeyword())
-        return item
+        # Update the query with the QC Analyses uids
+        qc_uids = map(api.get_uid, self.context.getQCAnalyses())
+        self.contentFilter.update({
+            "UID": qc_uids,
+            "portal_type": ["DuplicateAnalysis", "ReferenceAnalysis"],
+            "sort_on": "getId"
+        })
+
+        # Add Worksheet and QC Sample ID columns
+        new_columns = OrderedDict((
```

Comment:
I would do it the other way around. Put this into the `__init__` and flush it eventually when there are no QC analyses in `update`.